marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/EntitySearchControllerTest.java

```diff
@@ -57,30 +71,198 @@ void testCRUDOnSavedQuery() throws Exception {
+                assertRowsAndColumns(newLimit, totalColumns, response);
+                assertTrue(actualRowSet.contains(customer1.getHashCode()) || actualRowSet.contains(customer2.getHashCode()));
+            });
+    }
+
+    private void assertRowsAndColumns(int limit, int totalColumns, String response) {
+        int headerRow = 1;
+        int totalRows = limit + headerRow;
+
+        assertEquals(totalRows, response.chars().filter(i -> i == '\n').count() + 1);
+        assertEquals(totalColumns, response.substring(0, response.indexOf("\n")).chars().filter(i -> i == ',').count() + 1);
+    }
+
+    private void createCustomer(Customer customer) {
+        String customerEntityType = "Customer";
+
+        JSONDocumentManager mgr = getHubConfig().newFinalClient().newJSONDocumentManager();
+        ObjectNode customerProps = objectMapper.createObjectNode();
+        customerProps.put("customerId", customer.customerId);
+        customerProps.put("name", customer.name);
+        customerProps.put("customerNumber", customer.customerNumber);
+        customerProps.put("customerSince", customer.customerSince);
+
+        ObjectNode infoProp = objectMapper.createObjectNode();
+        infoProp.put("title", customerEntityType);
+        infoProp.put("version", "0.0.1");
+        infoProp.put("baseUri", "http://example.org/");
+
+        ObjectNode instanceProps = objectMapper.createObjectNode();
+        instanceProps.set("info", infoProp);
+        instanceProps.set(customerEntityType, customerProps);
+
+        ObjectNode instance = objectMapper.createObjectNode().set("instance", instanceProps);
+        ObjectNode envelope = objectMapper.createObjectNode().set("envelope", instance);
+
+        DocumentMetadataHandle metadata = new DocumentMetadataHandle()
+            .withCollections(customerEntityType)
+            .withPermission("data-hub-operator", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE);
+        mgr.write("/" + customerEntityType + customer.customerId + ".json", metadata, new JacksonHandle(envelope));
+    }
+
+    private Set<Integer> calculateHash(String csvData) throws IOException {
+        Set<Integer> rowSet = new HashSet<>();
+
+        try (BufferedReader reader = new BufferedReader(new StringReader(csvData))) {
+            String line = reader.readLine();
+            while (line != null) {
+                line = line.replaceAll(",", "");
+                rowSet.add(line.hashCode());
+                line = reader.readLine();
+            }
+        }
+
+        return rowSet;
+    }
+
+    private MultiValueMap<String, String> getRequestParams(int limit, String json) {
+        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("fileType", "csv");
+        params.add("limit", String.valueOf(limit));
+        params.add("queryDocument", json);
+
+        return params;
+    }
+
+    static class Customer {
```

Comment:
This can move to the testFixture source set in core DH too. Make it a regular public class though, with private fields and getters/setters, as we can use Jackson to generate JSON for it automatically, which will save some effort. We'll likely add more to this class so to make writing tests easier as we go.

Reply:
Will do.